kotlin/viewregistry-android/src/main/java/com/squareup/viewregistry/AbstractModalContainer.kt

```diff
@@ -0,0 +1,163 @@
+package com.squareup.viewregistry
+
+import android.app.Dialog
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import com.squareup.coordinators.Coordinator
+import com.squareup.coordinators.Coordinators
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.BehaviorSubject
+import kotlin.reflect.jvm.jvmName
+
+/**
+ * Base class for containers that show [IsModalContainerScreen.modals] in [Dialog]s
+ */
+abstract class AbstractModalContainer<M : Any>
+@JvmOverloads constructor(
+  context: Context,
+  attributeSet: AttributeSet? = null
+) : FrameLayout(context, attributeSet), HandlesBack {
+  private val base: View? get() = getChildAt(0)
+  private var dialogs: List<DialogRef<M>> = emptyList()
+  private val takeScreensSubs = CompositeDisposable()
+
+  private val attached = BehaviorSubject.create<Boolean>()
+
+  // Kind of a hack: use a Coordinator to create observable attached property.
+  // TODO should probably just drop Coordinators for RxBinding.
```

Comment:
Or architecture components? I think we can bridge directly from workflow core coroutines to lifecycle, maybe we can avoid rx entirely.